src/Features/Customer/Dashboard/ShowTab.js

```diff
@@ -0,0 +1,46 @@
+import React, { useState, useEffect } from 'react';
+
+import { Box, Grid } from '@material-ui/core';
+import { useDispatch } from 'react-redux';
+import { useHistory } from 'react-router';
+
+import MainTab from '../../../components/CardMenus/Tabs';
+import Loader from '../../../components/Loader';
+import NoDataFound from '../../../components/NoDataFilter';
+import { getCardData } from '../actions';
+import { GetCategories, GetItemsByVendor } from '../request';
+import { FirstTab } from './style';
+function ShowTab() {
+  const history = useHistory();
+  const params = new URLSearchParams(history.location.search);
+  const vendorId = params.get('id');
+  const [isTabshow, setTabShow] = useState(false);
+  const dispatch = useDispatch();
+  const { data: category } = GetCategories();
+  const { data: items, isFetching: isItemsFething } = GetItemsByVendor(vendorId);
+
+  useEffect(() => {
+    setTabShow(true);
+  }, [vendorId]);
+
+  useEffect(() => {
+    if (items) {
+      dispatch(getCardData(items));
```

Comment:
instead of dispatching the action in useEffect you can dispatch it onSuccess

Reply:
I think this is the right way to do it beacause fetch method of query does not give success callback option.. But there should be flag to check if the query is successfuk or not !!